app/main.js

```diff
@@ -144,7 +177,10 @@ function shutdown() {
 
 function upgrade(event, installerPath) {
   app.on('quit', () => {
-    shell.openItem(installerPath);
+    // shell.openItem doesn't reliably work from the app process, so run xdg-open directly
+    child_process.spawn('xdg-open', [installerPath], {
```

Comment:
It's called everywhere, but yesterday I factored this out and made it choose the command based on OS. That'll go up with my next set of changes.

Re: checking if it worked or failed, we can detect _some_ failure conditions (where the installer or package manager completely fails to launch, arguably the least likely scenario) by seeing if the process returns with an error status code. But we can't detect anything that happens after that, like getting that weird dialog box because of a user mismatch. But you're right, I'll add a check for an error code.